models/git/branch.go

```diff
@@ -378,3 +380,22 @@ func RenameBranch(ctx context.Context, repo *repo_model.Repository, from, to str
 
 	return committer.Commit()
 }
+
+// FindRecentlyPushedNewBranches return at most 2 new branches pushed by the user in 3 days which has no opened PRs created
+func FindRecentlyPushedNewBranches(ctx context.Context, repoID, userID int64) (BranchList, error) {
+	branches := make(BranchList, 0, 2)
+	subQuery := builder.Select("head_branch").From("pull_request").
+		InnerJoin("issue", "issue.id = pull_request.issue_id").
+		Where(builder.Eq{
+			"pull_request.head_repo_id": repoID,
+			"issue.is_closed":           false,
+		})
+	err := db.GetEngine(ctx).
+		Where("pusher_id=? AND is_deleted=?", userID, false).
+		And("updated_unix >= ?", time.Now().Add(-time.Hour*24*3).Unix()).
```

Comment:
Three days?
The other PR used 2 hours.
I'm not sure what a good timeout would be, but I'm rather for a relatively strict timeout somewhere between 2 and 6 hours. Afterwards, you probably forgot already about the changes you made…

Reply:
changed to 6 hours.